### ISSUE TITLE
Navigation block: submenus hidden by subsequent blocks (z-index issue) when Navigation is inside a Cover block

### ISSUE THREAD

**Issue by philnick206** [Automattic/wp-calypso#93675]
See [https://github.com/WordPress/gutenberg/issues/45353](https://github.com/WordPress/gutenberg/issues/45353) (WordPress.com fix blocked by this issue in core Gutenberg)

### Quick summary

Navigation blocks that are placed inside a Cover block and use submenus will have the submenus overlapped/hidden by subsequent blocks on the page.

![CleanShot 2024-08-19 at 10 11 44](https://github.com/user-attachments/assets/a31e0fc2-d008-4892-89b4-f7cc0bd230ac)

### Steps to reproduce

1. Create a Header that uses a Cover block and has a Navigation block inside (many common header patterns do this)
2. Add several menu items with submenu items to the menu
3. Add several Group blocks with content/patterns immediately after the Header/Cover block containing the Navigation

![CleanShot 2024-08-19 at 10 12 05](https://github.com/user-attachments/assets/48b6c25d-c302-428d-aa32-8d9390ac0876)

### What you expected to happen

The Navigation submenus should always be visible on desktop screens

### What actually happened

The Navigation submenus are hidden on desktop (ok on mobile in my testing)

### Impact

Most (> 50%)

### Available workarounds?

No but the platform is still usable

### If the above answer is "Yes...", outline the workaround.

*No response*

### Platform (Simple and/or Atomic)

*No response*

### Logs or notes

Reported in 8600675-zd-a8c

**Comment by jartes** [comment]
📌 REPRODUCTION RESULTS
- Tested on Simple – Replicated
- Tested on Atomic – Replicated
- Replicable outside of Dotcom – Yes

📌 FINDINGS/SCREENSHOTS/VIDEO
- This is coming from Core, and it's already reported here: https://github.com/WordPress/gutenberg/issues/53857
 
📌 ACTIONS
- Triaged
- Lowering priority from `High` to `Low` 
- Adding `Core Fix Needed` tag.

**Comment by ClassicRKR27** [comment]
@jartes I'm actually concerned that this issue has expanded beyond the `Cover` block.

I have a user here who is experiencing the same issue: 8627130-zd-a8c

But when I tried to remove it from the `Cover` block, that still wasn't enough - I had to completely remove it from the `Header` and all nested elements in order to get it to work, I've provided screenshots below:

<img width="1892" alt="Screenshot 2024-08-21 at 09 27 48" src="https://github.com/user-attachments/assets/9bfc898c-013a-4352-b579-61d3d6cf00f3">

<img width="1894" alt="Screenshot 2024-08-21 at 09 28 19" src="https://github.com/user-attachments/assets/9127d5e5-d997-4b15-9ecf-6596af5c9b14">

I'm sure it's related to the issue that you mentioned, but something seems to have changed and it's happening in more conditions now.

**Comment by ClassicRKR27** [comment]
Just wanted to update that @mxhassani was able to find a `CSS` solution (at least for the `Assembler` theme):

```
header.wp-block-template-part .wp-block-cover {
  overflow: visible;
  z-index: 99999;
}
```

**Comment by philnick206** [comment]
> Just wanted to update that @mxhassani was able to find a `CSS` solution (at least for the `Assembler` theme):
> 
> ```
> header.wp-block-template-part .wp-block-cover {
>   overflow: visible;
>   z-index: 99999;
> }
> ```

That workaround worked great for 8600675-zd-a8c

Thanks for sharing!

**Comment by ramonjd** [comment]
Hmmm, looks to be caused by the overflow rules on the cover block:

https://github.com/WordPress/gutenberg/blob/2f939a7f9047ed9b632ac4fb0d593d0cf9588ce9/packages/block-library/src/cover/style.scss#L10-L14

But those same rules are required for sticky positioning. 🤔 cc @andrewserong in case there's extra context here.

**Comment by andrewserong** [comment]
> But those same rules are required for sticky positioning. 🤔 cc @andrewserong in case there's extra context here.

I think they might have been added originally to handle border radius (so that the background is properly clipped to the radius of the border). The switch to `clip` instead of `hidden` was added for support for sticky positioning. The rule has been there for a long time now, so I'm not sure what a decent fix could be, unless there's a way for the rule to apply in more of a conditional way, only when it's needed, somehow? The hard thing there would be for us to know if border radius is being applied, given that could be set globally.

Unfortunately the main thing that comes to mind is that the Cover block isn't an ideal block to use for anything that would involve popovers (considering its issues with overflow here). Is there a reason why folks are using Cover blocks for their headers instead of a Group block? I.e. if it's about background images, the Group block now has pretty comprehensive background image support. If it's about being able to add an overlay color, though, then perhaps it highlights that it'd be good for Group blocks to be able to support overlays, I wonder?

**Comment by supernovia** [comment]
Noting this may be a duplicate of https://github.com/Automattic/wp-calypso/issues/75439 which is due to be fixed soon! 🥳

**Comment by supernovia** [comment]
This might be a duplicate or likeness of this issue:
https://github.com/Automattic/wp-calypso/issues/75439

**Comment by renatho** [comment]
> This might be a duplicate or likeness of this issue: [#75439](https://github.com/Automattic/wp-calypso/issues/75439)

I just checked this one, and it seems there's another issue here related to `overflow` in the cover block. So the original solution doesn't fix the submenus. It needs some more investigation to check the implications of removing the `overflow` and possible workarounds.

**Comment by getdave** [comment]
Related https://github.com/WordPress/gutenberg/issues/45353

**Comment by supernovia** [comment]
https://a8c.zendesk.com/agent/tickets/9237617

**Comment by cat-og** [comment]
Another report here: 9306074-zd-a8c

 Theme: Casthub: https://wordpress.com/theme/casthub/

They're on the Personal plan, so I recommended removing those Header blocks from the Cover block, and setting the image as the background image on one of the Group blocks as a work-around.
